src/metabase/query_processor/middleware/parameters/mbql.clj

```diff
@@ -17,9 +17,9 @@
   convert the parameters to integers or floating-point numbers."
   [param-type param-value, field-id :- su/IntGreaterThanZero]
   (cond
-    ;; for `id` type params look up the base-type of the Field and see if it's a number or not. If it *is* a number
-    ;; then recursively call this function and parse the param value as a number as appropriate.
-    (and (= param-type :id)
+    ;; for `id` or `category` type params look up the base-type of the Field and see if it's a number or not.
+    ;; If it *is* a number then recursively call this function and parse the param value as a number as appropriate.
+    (and (or (= param-type :id) (= param-type :category))
```

Comment:
In Clojure a common idiom for this situation is to do 

```clj
(#{:id :category} param-type)
```

`#{:id :category}` = a set literal 
passing an arg to a set in clojure returns the arg if it is in the set or `nil` if it is not.

Reply:
Thanks for the tip! I am new to Clojure so I don't know the idioms yet. I just updated the PR with that change.